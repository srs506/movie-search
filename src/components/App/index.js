import * as React from 'react';
import Table from '../Table';
import './index.css';

const PATH_BASE = `https://api.themoviedb.org/3`;
const PATH_DISCOVER = '/discover';
const PATH_MOVIE = '/movie';
const PATH_CONFIGURATION = '/configuration';
const PARAM_API = 'api_key=';
const API_KEY = '1835ac896829a73f680b0d9f6094d57a';
const PARAM_VOTE_AVG_GTE = 'vote_average.gte=';
const DEFAULT_VOTE_AVG_GTE = '6.0';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      movieList: null,
      voteAverageGte: DEFAULT_VOTE_AVG_GTE,
      imageBaseUrl: null,
      posterSizes: null
    };

    this.fetchMovies = this.fetchMovies.bind(this);
    this.fetchConfiguration = this.fetchConfiguration.bind(this);
    this.setConfiguration = this.setConfiguration.bind(this);
    this.setMovieList = this.setMovieList.bind(this);
  }

  setMovieList(result) {
    this.setState({
      movieList: result
    });
  }

  setConfiguration(result) {
    console.log(result.images);
    const { secure_base_url, poster_sizes } = result.images;

    this.setState({
      imageBaseUrl: secure_base_url,
      posterSizes: poster_sizes
    });
  }

  fetchMovies(voteAverage, page = 0) {
    fetch(
      `${PATH_BASE}${PATH_DISCOVER}${PATH_MOVIE}?${PARAM_API}${API_KEY}&${PARAM_VOTE_AVG_GTE}${voteAverage}`
    )
      .then(response => response.json())
      .then(result => this.setMovieList(result))
      .catch(e => e);
  }

  fetchConfiguration() {
    fetch(`${PATH_BASE}${PATH_CONFIGURATION}?${PARAM_API}${API_KEY}`)
      .then(response => response.json())
      .then(result => this.setConfiguration(result))
      .catch(e => e);
  }

  render() {
    const { movieList, imageBaseUrl, posterSizes } = this.state;
    const list = (movieList && movieList.results) || [];

    return (
      <div className="container">
        <div className="main">
          <Table
            list={list}
            imageBaseUrl={imageBaseUrl}
            posterSize={(posterSizes && posterSizes[0]) || null}
          />
        </div>
      </div>
    );
  }

  componentDidMount() {
    const { voteAverageGte } = this.state;
    this.setState({ voteAverage: voteAverageGte });

    this.fetchConfiguration();
    this.fetchMovies(voteAverageGte);
  }
}

export default App;
