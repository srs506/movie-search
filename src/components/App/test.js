import React from 'react';
import ReactDOM from 'react-dom';
import App from './index';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

test('has a valid snapshot', () => {
  const component = renderer.create(<App />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
