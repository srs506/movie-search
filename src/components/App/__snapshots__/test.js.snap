// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has a valid snapshot 1`] = `
<div
  className="container"
>
  <div
    className="main"
  >
    <form
      onSubmit={[Function]}
    >
      <fieldset>
        <label
          htmlFor="voteAverageInput"
        >
          Vote Average
        </label>
        <input
          max="10"
          min="0"
          onChange={[Function]}
          step=".1"
          type="number"
          value="6.0"
        />
        <input
          className="button-primary"
          id="voteAverageInput"
          type="submit"
          value="Submit"
        />
      </fieldset>
    </form>
    <table>
      <tbody />
    </table>
  </div>
  <div
    className="footer"
  >
    <img
      alt="Powered by The Movie DB"
      height="60px;"
      src="powered-by-rectangle-blue.svg"
    />
  </div>
</div>
`;
